src/main/java/com/google/gcp_variant_transforms/library/SchemaGeneratorImpl.java

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 Google LLC
+
+package com.google.gcp_variant_transforms.library;
+
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.common.collect.ImmutableList;
+import htsjdk.variant.vcf.VCFHeader;
+
+
+
+/**
+ * Service to create BigQuery Schema from VCFHeader 
+ */
+public class SchemaGeneratorImpl implements SchemaGenerator {
+
+  public static class FieldDescriptionConstants {
+    public static final String REFERENCE_NAME = "Reference name.";
+    public static final String START_POSITION = "Start position. Corresponds to the first " +
+                                                "base of the string of reference bases.";
+    public static final String END_POSITION = "End position. Corresponds to the first base " +
+                                              "after the last base in the reference allele.";
+    public static final String REFERENCE_BASES = "Reference bases.";
+    public static final String NAMES = "Variant names (e.g. RefSNP ID).";
+    public static final String QUALITY = "Phred-scaled quality score (-10log10 prob(call " +
+                                         "is wrong)). Higher values imply better quality.";
+    public static final String FILTER = "List of failed filters (if any) or \"PASS\" " +
+                                        "indicating the variant has passed all filters.";
+    public static final String CALLS = "One record for each call.";
+    public static final String CALLS_SAMPLE_ID = "Unique ID (type INT64) assigned to each " +
+                                                 "sample. Table with `__sample_info` suffix " +
+                                                 "contains the mapping of sample names (as read" +
+                                                 "from VCF header) to these assigned IDs.";
+    public static final String CALLS_GENOTYPE = "Genotype of the call. \"-1\" is used in cases " +
+                                                "where the genotype is not called.";
+    public static final String CALLS_PHASESET = "Phaseset of the call (if any). \"*\" is used " +
+                                                "in cases where the genotype is phased, but no " +
+                                                "phase set (\"PS\" in FORMAT) was specified.";
+  }
+  
+  public Schema getSchema(VCFHeader vcfHeader){
+    ImmutableList<Field> schemaFields = getFields(vcfHeader);
+    Schema schema = Schema.of(schemaFields);
+    return schema;
+  }
+
+  public ImmutableList<Field> getFields(VCFHeader vcfHeader){
+    ImmutableList.Builder<Field> fields = new ImmutableList.Builder<Field>();
+    Field fieldReferenceName = Field.newBuilder(
```

Comment:
If we also add two more maps for `Type` and `Mode`, here we would only need to look up those values for each column.

I wonder what @tneymanov or @BoscoSuen think?

Reply:
Let me try to understand, you'd like to have a 2 maps:
```
FieldNameToType = {
  "reference_name" -> STRING. // or more programatically, ColumnKeyConstants.REFERENCE_NAME -> STRING
  "start_position" -> INT64
  "reference_bases" -> STRING
  ...
}

FieldNameToMode {
  "reference_name" -> NULLABLE
  "START_POS" -> NULLABLE  // or more programatically, ColumnKeyConstants.START_POSITION -> NULLABLE
  "REF_BASES" -> REPEATED
  ...
}
```

And then in code, have something like:
```
Field fieldStartPos = Field.newBuilder(
  ColumnKeyConstants.REFERENCE_NAME, FieldNameToType.get(ColumnKeyConstants.REFERENCE_NAME))
  .setMode(FieldNameToMode.get(ColumnKeyConstants.REFERENCE_NAME)
```
?

I can see that, but in that case, why can't we also create a mapping from FieldName to Description and then we would be able to loop through the list of constants, and add the fields that way in a single block, by just getting values from mapping.

ie
```
for constant in constants:
  field = Field(constant, FieldNameToType(constant))
      .setMode(FieldNameToMode(constant)
      ...
```

WDYT

Reply:
> why can't we also create a mapping from FieldName to Description

I also suggested that in a separate comment.
So I guess we agree 3 maps is a better way to hardcode the information.

Reply:
What Tural is saying makes sense for the constant columns. The loop with constants fields would work for the first ~7 columns until we get to the Calls field (a record that has subfields). Maybe for this we could include an if statement in the for loop to see when we get to the call record, and then similarly loop through each call sub-field. Does this make sense? 

After these steps we would have to use the BigQuery constants for name, mode and type since these dynamic fields won't have an entry in our maps.

If we create the maps then it makes a lot of sense to create a separate utils for Schema generation--agreed?